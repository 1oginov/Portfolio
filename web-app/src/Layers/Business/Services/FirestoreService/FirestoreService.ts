import firebase from 'firebase';

import {
  FIRESTORE_POSITIONS_COLLECTION, FIRESTORE_USERS_COLLECTION,
} from 'Layers/Infrastructure/Constants/FirestoreConstants/FirestoreConstants';

export const getPositionsCollectionPath = (userId: string): string => (
  `${FIRESTORE_USERS_COLLECTION}/${userId}/${FIRESTORE_POSITIONS_COLLECTION}`
);

export const getPositionsCollection = (
  firebaseInstance: firebase.app.App,
  userId: string,
): firebase.firestore.CollectionReference => (
  firebaseInstance.firestore()
    .collection(FIRESTORE_USERS_COLLECTION)
    .doc(userId)
    .collection(FIRESTORE_POSITIONS_COLLECTION)
);

export const getPositionDocument = (
  firebaseInstance: firebase.app.App,
  userId: string,
  positionId: string,
): firebase.firestore.DocumentReference => getPositionsCollection(firebaseInstance, userId).doc(positionId);
