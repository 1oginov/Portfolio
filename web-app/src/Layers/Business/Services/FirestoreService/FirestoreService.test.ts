import { getPositionsCollectionPath } from './FirestoreService';

describe('getPositionsCollectionPath', () => {
  it('matches snapshot', () => {
    expect(getPositionsCollectionPath('userId')).toMatchSnapshot();
  });
});
