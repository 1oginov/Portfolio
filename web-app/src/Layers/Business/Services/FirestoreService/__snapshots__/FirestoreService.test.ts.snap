// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPositionsCollectionPath matches snapshot 1`] = `"users/userId/positions"`;
